src/main/groovy/edu/oregonstate/mist/metaxe/XEAppDAO.groovy

```diff
@@ -0,0 +1,123 @@
+package edu.oregonstate.mist.metaxe
+
+import com.fasterxml.jackson.core.JsonParseException
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.databind.JsonMappingException
+import javax.ws.rs.core.UriBuilder
+
+@groovy.transform.TypeChecked
+class XEAppDAO {
+    private URI esUrl
+    private ObjectMapper mapper = new ObjectMapper()
+
+    XEAppDAO(URI elasticsearchUrl) {
+        this.esUrl = elasticsearchUrl
+    }
+
+    ESResult getById(String id) {
+        def builder = UriBuilder.fromUri(this.esUrl)
+        def url = builder.path('{id}').build(id).toURL()
+
+        println "get by id: ${url}"
+
+        InputStream jsonStream
+        try {
+            jsonStream = url.newInputStream()
+        } catch (FileNotFoundException e) {
+            // 404 -> return null
```

Comment:
Which information would you like logged? The ElasticsearchExceptions will be caught and logged by GenericExceptionMapper. This is the only error case which doesn't throw an exception, but returning null here will ultimately cause a 404 response to be sent to the client, which will show up in the request logs.